api/v1beta1/machine_webhook.go

```diff
@@ -60,6 +64,10 @@ func (m *Machine) Default() {
 		normalizedVersion := "v" + *m.Spec.Version
 		m.Spec.Version = &normalizedVersion
 	}
+
+	if m.Spec.NodeDeletionTimeout == nil {
+		m.Spec.NodeDeletionTimeout = &metav1.Duration{Duration: defaultNodeDeletionTimeout}
```

Comment:
Does it make sense to use `// kubebuilder:default="10s"` instead? That would also allow omitting the pointer, but the default would be applied on MachineDeployments, MachineSets and MachinePools as well, and not just on Machines.

In general, what's preferred here? The spec default or custom logic in the webhook?

Reply:
Webhook is probably better